extractor.go

```diff
@@ -346,30 +346,34 @@ func extract(ctx context.Context, t Target, dst string, src archiveWalker, cfg *
 			}
 
 			// open file inm archive
-			fin, err := ae.Open()
-			if err != nil {
-				return handleError(cfg, td, "failed to open file", err)
-			}
-			defer fin.Close()
+			err, fileCreated := func() (error, bool) {
+				fin, err := ae.Open()
+				if err != nil {
+					return handleError(cfg, td, "failed to open file", err), false
+				}
```

Comment:
Previously if there was an error from `ae.Open()` we always returned irrespective of whether `handleError` returned `nil` but with this change we will continue with the next file in the archive (when `handleError` returns nil). Just curious, is that the expected behaviour?

Reply:
good observation, missed it. But I think it should be more aligned with what is expected from `ContinueOnError` option, but maybe there was another reason to do it that way, @NodyHub any comments?

Reply:
Oh wow, great observation @kartheek-hc & thanks for pinging @vladimir-buzuev !!

@vladimir-buzuev you are 💯 correct with the point that the behaviour should align with `ContinueOnError`. The change you made in your PR aligns better then before with that option, which led before this PR to a potential miss // unexpected behaviour.